pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -692,6 +705,7 @@ func (p *PriorityQueue) determineSchedulingHintForInFlightPod(logger klog.Logger
 		}
 		return queueSkip
 	}
+	p.activeQLock.RUnlock()
```

Comment:
This may lead to deadlock if we return early.

Reply:
You're right, thanks for noticing this